pkg/catalog/catalog.go

```diff
@@ -8,17 +8,38 @@ import (
 
 	"github.com/golang/glog"
 	imagev1 "github.com/openshift/api/image/v1"
+	"github.com/openshift/odo/pkg/log"
 	"github.com/openshift/odo/pkg/occlient"
+	"github.com/openshift/odo/pkg/preference"
 	"github.com/openshift/odo/pkg/util"
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// DevfileRegistries contains the links of all devfile registries
-var DevfileRegistries = []string{
-	"https://raw.githubusercontent.com/elsony/devfile-registry/master",
-	"https://che-devfile-registry.openshift.io/",
+// GetDevfileRegistries gets devfile registries from preference file
+func GetDevfileRegistries(registryNameList []string) (map[string]string, error) {
+	devfileRegistries := make(map[string]string)
+
+	cfg, err := preference.New()
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.OdoSettings.RegistryList != nil {
+		for _, registry := range *cfg.OdoSettings.RegistryList {
+			if len(registryNameList) != 0 && registryNameList[0] == registry.Name {
+				devfileRegistries[registry.Name] = registry.URL
+				break
```

Comment:
please add a comment why are we breaking here instead returning all the values

Reply:
I update the logic in new commit and add corresponding comments